OpenOversight/app/main/views.py

```diff
@@ -1304,3 +1300,99 @@ def dispatch_request(self, *args, **kwargs):
     '/officer/<int:officer_id>/description/<int:obj_id>/delete',
     view_func=description_view,
     methods=['GET', 'POST'])
+
+
+# This API only applies to links attached to officer profiles, not links
+# attached to incidents.
+class OfficerLinkApi(ModelView):
+    model = Link
+    model_name = 'link'
+    form = OfficerLinkForm
+    department_check = True
+
+    @property
+    def officer(self):
+        if not hasattr(self, '_officer'):
+            self._officer = db.session.query(Officer).filter_by(id=self.officer_id).one()
+        return self._officer
+
+    @login_required
+    @ac_or_admin_required
+    def new(self, form=None):
+        if not current_user.is_administrator and current_user.ac_department_id != self.officer.department_id:
+            abort(403)
+        if not form:
+            form = self.get_new_form()
+            if hasattr(form, 'creator_id') and not form.creator_id.data:
+                form.creator_id.data = current_user.get_id()
+
+        if form.validate_on_submit():
```

Comment:
Can you add an `else` to catch if there are errors and log them, in the spirit of https://github.com/lucyparsons/OpenOversight/issues/786 ?

Reply:
The ModelView class it inherits from is written in the same way, so I wouldn't want to break that pattern here without doing it elsewhere, to keep things consistent. And I think it would be best to come to consensus on a solution for #786 and apply it in a separate PR.